src/bramble/client/ProjectLimiter.js

```diff
@@ -0,0 +1,94 @@
+define(function() {
+"use strict";
+    var ProjectLimiter = {};
+
+    var maxProjectState = {
+            fileSize: 0, // max size of any file in bytes. NOT USED now.
+            fileCount: 0, // total number of files you can have in a project
+            projectSize: 0// total number of bytes for all files in this project
+    };
+    var currentProjectState = {
+        numOfFiles: 0 ,
+        projectSize: 0
+    };
+
+    ProjectLimiter.initMaxLimits =  function (projectState){
+        maxProjectState = projectState;
+        console.log(projectState);
+    };
+   function _checkLimits(){
+        if(currentProjectState.numOfFiles >= maxProjectState.fileCount || currentProjectState.projectSize >= maxProjectState.projectSize){
+            console.log("Maximum project size or file count is exceeded!\nCurrent number of files: "+currentProjectState.numOfFiles 
+            + "\nCurrent project size: " + currentProjectState.projectSize + "\nMaximum project file count: "
+            + maxProjectState.fileCount + "\nMaximum project size: " + maxProjectState.projectSize);
+        }
+    };
+
+    ProjectLimiter.newFile = function(){
+            currentProjectState.numOfFiles +=1;
```

Comment:
My naming of variables should have confused you. Root parameter in the checkLimits function is not actually a root, rather a pathname to a file we are working with right now. I never walk the whole file tree.
I see why it is confusing: I am passing a recursive option to the ls call, which is not really needed, since I only deal with the files anyways.